_source/_posts/2022-03-29-github-actions-graalvm.adoc

```diff
@@ -0,0 +1,490 @@
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+
+      - name: Configure pagefile
+        uses: al-cheb/configure-pagefile-action@v1.2
+        with:
+          minimum-size: 10GB
+          maximum-size: 12GB
+
+      - name: Set up pagefile
+        run: |
+          (Get-CimInstance Win32_PageFileUsage).AllocatedBaseSize
+      - name: mvnw --version
+        run: mvnw --version
+        shell: cmd
+
+      - name: Maven resolve
+        run: mvnw -B -ntp dependency:resolve-plugins
+        shell: cmd
+
+      - name: Build native images
+        run: mvnw -B -ntp package -Pnative,prod -DskipTests
+        shell: cmd
+
+      - name: Archive binary
+        uses: actions/upload-artifact@v2
+        with:
+          name: flickr-${{ matrix.os }}-x86_64.exe
+          path: target/native-executable.exe
+
+      - name: Rename binary
+        run: move target/native-executable.exe target/flickr2-${{ runner.os }}-${{ github.event.release.tag_name }}-x86_64.exe
+      - name: Upload release
+        uses: alexellis/upload-assets@0.3.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          asset_paths: '["target/flickr2-${{ runner.os }}*"]'
+----
+{% endraw %}
+
+Add this file to your `main` branch.
+
+[source,shell]
+----
+git add .
+git commit -m "Add native artifacts to release"
+git push origin main
+----
+
+CAUTION: I recently tried to use `windows-latest` instead of `windows-2019` and I https://github.com/graalvm/setup-graalvm/issues/6#issuecomment-1058328963[ran out of disk space].
+
+=== Publish a release on GitHub
+
+Open your repository's page in your favorite browser and click on *Create a new release*. Create a new `v0.0.1` tag, title the release `v0.0.1`, and add some fun text in the description. Click **Publish release**.
+
+image::{% asset_path 'blog/github-actions-graalvm/release.png' %}[alt=Restore v0.0.1 - Giddyup!,width=800, align=center]
+
+Click the **Actions** tab to watch your release execute. I'd like to warn you though, it's gonna take a while! My https://twitter.com/mraible/status/1498471457638293507[first successful release] took just under an hour.
+
+- macOS: 31m 30s
+- Linux: 33m 50s
+- Windows: 59m 45s
+
+However, I think you'll be pleased with the results. 🤠
+
+image::{% asset_path 'blog/github-actions-graalvm/release-with-artifacts.png' %}[alt=Released with native binaries attached,width=800,align=center]
+
+TIP: If your builds fail, you can delete the tag for the release by running `git push origin :v0.0.1`. Your release will then become a draft, and you can easily create the release again using the GitHub UI.
+
+== Run your released binaries locally
+
+If you were to download these binaries from GitHub and try to run them locally, you'd get failures because they can't connect to instances of Keycloak or PostgreSQL.
+
+To start up a PostgreSQL database for it to talk to, you can run the following command from your `flickr2` directory.
```

Comment:
What's the "it" refer to in "for it to talk to" ? Unable to guess it.

Reply:
it == the JHipster app or your full-stack Java app, depending on how SEO-aggressive we want to be.

Reply:
@havi Do you think I should change "it" to something else?